### PR TITLE
feat(auth): add hub events

### PR DESCRIPTION
**Pull request by israx**
#### Description of changes

 - adds hub events on the `signInWithRedirect` API which is not able to resolve with a result. 
 - adds hub events on every `refreshToken` action. This behavior happens internally and it notifies when tokens are refreshed or if there is an error while refreshing.

#### Checklist

- [x] PR description included
- [ ] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by elorzafe — CHANGES_REQUESTED** [review]
Can we have a unit test?

**Review by jimblanc — DISMISSED** [review]

File: packages/core/src/Hub/types/AuthTypes.ts [hunk 1]
```diff
@@ -1,10 +1,12 @@
...
 // SPDX-License-Identifier: Apache-2.0
 
-// TODO Need to update types of data
 export type AuthHubEventData =
-	| { event: 'signInWithRedirect' } // Used when an oauth flow is done
-	| { event: 'tokenRefresh' } // used when a token is refreshed
-	| { event: 'tokenRefresh_failure' } // used when the refresh of tokens failed
-	| { event: 'customOAuthState' }
-	| { event: 'customState_failure' };
+	/** Dispatched when a user signs in with a oauth provider such as Google.*/
```
Comment (kvramyasri7) [inline packages/core/src/Hub/types/AuthTypes.ts:5]: very nit : 
```suggestion
	/** Dispatched when a user signs in with an oauth provider such as Google.*/
```

File: packages/auth/__tests__/providers/cognito/hub.test.ts [hunk 2]
```diff
@@ -0,0 +1,97 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
```
Comment (kvramyasri7) [inline packages/auth/__tests__/providers/cognito/hub.test.ts:2]:
```suggestion
// SPDX-License-Identifier: Apache-2.0

```

**Review by kvramyasri7 — APPROVED** [review]
Nothing blocking. LGTM. Thanks for doing this @israx

**Review by jimblanc — APPROVED** [review]

**Review by jimblanc — APPROVED** [review]

**Review by kvramyasri7 — APPROVED** [review]

**Review by elorzafe — APPROVED** [review]

**Pull request merged**